linkcheck/logger/csvlog.py

```diff
@@ -19,7 +19,10 @@
 """
 import csv
 import os
-from io import BytesIO
+try:
+    from StringIO import StringIO
```

Comment:
I would make that `from cStringIO import StringIO`, but it probably doesn't matter.

Reply:
That's on me. It was cStringIO but I changed it because I had found problems elsewhere (will post later) because cStringIO can't handle all Unicode. I think we probably should change this one back for now.

Reply:
Another option: Python 2 now has io.StringIO, so we could use the same for both?

Reply:
This is passed to csv.writer, right?  csv is a nasty module: it wants native strings (i.e. bytes on Python 2 and str on Python 3).  You can't write bytes to io.StringIO() on Python 2.  You might get away with using io.BytesIO() on Python 2 and io.StringIO() on Python 3.

I'm very surprised to hear that you've discovered a difference between StringIO and cStringIO.  IIRC both handle a mixture of bytes and unicode as long as that mixture is ASCII-only, but as soon as you stray off the safe path, both explode.

Reply:
I've put from `cStringIO import StringIO` back.